sesman/session.c

```diff
@@ -420,15 +349,51 @@ wait_for_xserver(int display)
 }
 
 /******************************************************************************/
+static int
+session_start_chansrv(char *username, int display)
+{
+    struct list *chansrv_params;
+    char exe_path[262];
+    int cspid;
+
+    cspid = g_fork();
+    if (cspid == 0)
+    {
+        chansrv_params = list_create(); 
+        chansrv_params->auto_free = 1;
+
+        /* building parameters */
+        g_snprintf(exe_path, 261, "%s/xrdp-chansrv", XRDP_SBIN_PATH);
+
+        list_add_item(chansrv_params, (intptr_t) g_strdup(exe_path));
+        list_add_item(chansrv_params, 0); /* mandatory */
+
+        env_set_user(username, 0, display,
+                     g_cfg->session_variables1,
+                     g_cfg->session_variables2);
+
+        /* executing chansrv */
+        g_execvp(exe_path, (char **) (chansrv_params->items));
+        /* should not get here */
+        log_message(LOG_LEVEL_ALWAYS, "error starting chansrv "
+                    "- user %s - pid %d", username, g_getpid());
+        list_delete(chansrv_params);
+        g_exit(1);
+    }
+    return cspid;
+}
+
+/******************************************************************************/
 /* called with the main thread */
 static int
-session_start_fork(tbus data, tui8 type, struct SCP_SESSION *s)
+session_start_fork(tbus data, tui8 type, struct SCP_CONNECTION *c,
+                   struct SCP_SESSION *s)
 {
     int display = 0;
     int pid = 0;
     int wmpid = 0;
-    int pampid = 0;
     int xpid = 0;
+    int cspid = 0;
```

Comment:
I would appreciate if we use longer names for new variables. It's really helpful. "cs" can be anything: chanserv, client session, critical section, whatever. Also, underscores increase readability.